### PR TITLE
etcd: add --provenance=false during etcd multi-arch build

### LINKED ISSUE #31154: [Failing Test] post-kubernetes-push-image-etcd
### Which jobs are failing?

master-informing:
- post-kubernetes-push-image-etcd

### Which tests are failing?

Overall job

### Since when has it been failing?

First failure: https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/post-kubernetes-push-image-etcd/1718020233660207104

### Testgrid link

https://testgrid.k8s.io/sig-release-master-informing#post-kubernetes-push-image-etcd

### Reason for failure (if possible)

```
ERROR: (gcloud.builds.submit) build 0bbd94cb-0c8b-41f0-9631-53590a87a4c7 completed with status "FAILURE"
```

Some additional suspicions:
```
/bin/bash: line 1: ./build.sh: No such file or directory
done
install: can't stat '/workspace/tmp.BbIfCf/etcd': No such file or directory
install: can't stat '/workspace/tmp.BbIfCf/etcdctl': No such file or directory
install: can't stat '/workspace/tmp.ilEfpe/etcd': No such file or directory
install: can't stat '/workspace/tmp.ilEfpe/etcdctl': No such file or directory

cp: cannot stat 'bin/arm64/etcd*': No such file or directory
....
```

### Relevant SIG(s)

/sig k8s-infra

### PR DESCRIPTION
**Pull request by pacoxu**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
See https://docs.docker.com/build/attestations/

#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes/test-infra/issues/31154

THis is docker new behavior change for the supply chain security. Not sure if we can fix it like this. 

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: cluster/images/etcd/Makefile [hunk 1]
```diff
@@ -192,6 +192,7 @@ endif
 	# And build the image
 	docker buildx build \
 		--pull \
+		 --sbom=false --provenance=false \
```
Comment (saschagrunert) [inline cluster/images/etcd/Makefile:195]: nit:
```suggestion
		--sbom=false \
		--provenance=false \
```

**Review by mzaian — APPROVED** [review]
Hello @pacoxu,
Thanks for the fix.

File: cluster/images/etcd/Makefile (hunk 1 above)
Comment (pacoxu) [reply cluster/images/etcd/Makefile]: I check some other examples:
- `--provenance` is necessary
- I removed the sbom part.

**Review by saschagrunert — APPROVED** [review]

**Review by serathius — APPROVED** [review]

**Review by liangyuanpeng — APPROVED** [review]

**Review by r0b2g1t — APPROVED** [review]

**Comment by dims** [comment]
/milestone v1.29

discussion thread in slack: https://kubernetes.slack.com/archives/C2C40FMNF/p1698844181604919?thread_ts=1698666130.888699&cid=C2C40FMNF

**Pull request merged**